src/Lecture2.hs

```diff
@@ -70,7 +79,37 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt i list
+   | i < 0               = (Nothing, list)
+   | null list           = (Nothing, list)
+   | i == 0              = (Just (head list), tail list)
```

Comment:
I see that you use unsafe functions `head` and `tail` here 🗿
In that case, it's better to pattern match on lists